pkg/controller/atlasproject/project_settings.go

```diff
@@ -0,0 +1,73 @@
+package atlasproject
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"reflect"
+
+	v1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
+)
+
+func ensureProjectSettings(ctx *workflow.Context, projectID string, project *v1.AtlasProject) (result workflow.Result) {
+	if result = syncProjectSettings(ctx, projectID, project); !result.IsOk() {
+		ctx.SetConditionFromResult(status.ProjectSettingsReadyType, result)
+		return result
+	}
+
+	if areProjectSettingsEmpty(project.Spec.Settings) {
+		ctx.UnsetCondition(status.ProjectSettingsReadyType)
+		return workflow.OK()
+	}
+
+	ctx.SetConditionTrue(status.ProjectSettingsReadyType)
+	return workflow.OK()
+}
+
+func syncProjectSettings(ctx *workflow.Context, projectID string, project *v1.AtlasProject) workflow.Result {
+	spec := project.Spec.Settings
+
+	atlas, err := fetchSettings(ctx, projectID)
+	if err != nil {
+		return workflow.Terminate(workflow.ProjectSettingsReady, err.Error())
+	}
+
+	if !areSettingsInSync(atlas, spec) {
+		if err := patchSettings(ctx, projectID, spec); err != nil {
+			return workflow.Terminate(workflow.ProjectSettingsReady, err.Error())
+		}
+	}
+
+	return workflow.OK()
+}
+
+func areProjectSettingsEmpty(settings *v1.ProjectSettings) bool {
+	return settings == nil
+}
+
+func areSettingsInSync(atlas, spec *v1.ProjectSettings) bool {
+	if areProjectSettingsEmpty(spec) {
+		return true
+	}
+
+	return reflect.DeepEqual(atlas, spec)
+}
+
+func patchSettings(ctx *workflow.Context, projectID string, spec *v1.ProjectSettings) error {
+	path := fmt.Sprintf("api/atlas/v1.0/groups/%s/settings", projectID)
+	_, err := ctx.Client.NewRequest(context.Background(), http.MethodPatch, path, spec)
```

Comment:
Should we consider visit again this call (and other already in place) and decorate atlas client with not yet implemented endpoints? we avoid spread custom direct call to atlas api in the controllers and centralize this on the client (where it should be)

Reply:
Not sure. This PATCH call override is only necessary because there is a mistake in the lib implementation so as soon as this mistake is gone we can swap it for a regular lib call. So at this point I'm actually happy it is contained in this single module and the rest of the project doesn't know that this walk-around is being used.

I'm thinking of personally making a PR to the lib so it shouldn't stay like this for long:)

Reply:
Not sure. This PATCH call override is only necessary because there is a mistake in the lib implementation so as soon as this mistake is gone we can swap it for a regular lib call. So at this point I'm actually happy it is contained in this single module and the rest of the project doesn't know that this walk-around is being used.

I'm thinking of personally making a PR to the lib so it shouldn't stay like this for long:)